scripts/ai/controllers/ImplementController.lua

```diff
@@ -4,6 +4,18 @@ ImplementController = CpObject()
 function ImplementController:init(vehicle, implement)
     self.vehicle = vehicle
     self.implement = implement
+    self.overwrittenFunctions = {}
+    self.additionalAIEvents = {}
+end
+
+function ImplementController:delete()
+    self:unregisterOverwrittenFunctions()
+    self:unregisterAIEvents()
+end
+
+function ImplementController:setAllStaticParameters(settings,fillLevelManager)
+    self.settings = settings
+    self.fillLevelManager = fillLevelManager

```

Comment:
We set this but then don't seem to be using for everything. We have to make a decision how we want to handle fill levels. We can move it to the controllers, but not sure how to handle if the fill type is not on the same implement, or on the vehicle directly etc (in other words, fill levels are vehicle global and not necessarily local to the specific implement, therefore something like the FillLevelManager may make sense.

Reply:
I understand your idea behind the FillLevelManager. My idea was to seperate these for now, as we only need to know if we can continue the work for now and this is diffrent for each implement. For example the possible forage wagon needs to ignore the additional filltype (silage additives) or the sowing machine might need to ignore the fertilizer depending on the setting for it.